src/test/java/com/rultor/agents/github/ClosePullRequestTest.java

```diff
@@ -0,0 +1,156 @@
+        final Issue issue = repo.issues().create("", "");
+        final Profile profile = new Profile.Fixed(
+            new XMLDocument(
+                StringUtils.join(
+                    "<p><entry key='merge'>",
+                    "<entry key='rebase'>true</entry>",
+                    "</entry></p>"
+                )
+            )
+        );
+        final Agent agent = new ClosePullRequest(profile, repo.github());
+        final Talk talk = ClosePullRequestTest.createTalk(repo, issue);
+        agent.execute(talk);
+        MatcherAssert.assertThat(
+            new Issue.Smart(issue).state(), Matchers.is("closed")
+        );
+    }
+
+    /**
+     * ClosePullRequest leaves issue open if no rebase mode.
+     * @throws Exception If error
+     */
+    @Test
+    public void leavesPullRequestOpenWhenNoRebaseMode() throws Exception {
+        final Repo repo = new MkGithub().randomRepo();
+        final Issue issue = repo.issues().create("", "");
+        final Profile profile = new Profile.Fixed(
+            new XMLDocument(
+                StringUtils.join(
+                    "<p> <entry key='merge'>",
+                    "  <entry key='rebase'>false</entry>",
+                    "</entry> </p>"
+                )
+            )
+        );
+        final Agent agent = new ClosePullRequest(profile, repo.github());
+        final Talk talk = ClosePullRequestTest.createTalk(repo, issue);
+        agent.execute(talk);
+        MatcherAssert.assertThat(
+            new Issue.Smart(issue).state(), Matchers.is("open")
+        );
+    }
+
+    /**
+     * ClosePullRequest can comment issue with explanation.
+     * @throws Exception If error
+     */
+    @Test
+    public void commentsToIssueWithExplanation() throws Exception {
+        final Repo repo = new MkGithub().randomRepo();
+        final Issue issue = repo.issues().create("", "");
+        final Profile profile = new Profile.Fixed(
+            new XMLDocument(
+                StringUtils.join(
+                    "<p><entry key='merge' >",
+                    "<entry key='rebase' >true</entry>",
+                    " </entry> </p>"
+                )
+            )
+        );
+        final Agent agent = new ClosePullRequest(profile, repo.github());
+        final Talk talk = ClosePullRequestTest.createTalk(repo, issue);
+        agent.execute(talk);
+        final Comment.Smart comment = new Comment.Smart(
+            new Issue.Smart(issue).comments().get(1)
+        );
+        MatcherAssert.assertThat(
+            comment.body(), Matchers.containsString(
+                "Closed manually because of rebase mode"
+            )
+        );
+    }
+
+    /**
+     * Creates talk with href, github repository, github issue.
+     * @param repo Repository
+     * @param issue Issue
+     * @return Talk
+     * @throws IOException if errors
+     */
+    private static Talk createTalk(final Repo repo, final Issue issue)
```

Comment:
@vkuchyn this method can be named as `talk`
Please, see these chapters for the explanation http://www.yegor256.com/images/books/elegant-objects/seven-pages.pdf